runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -845,14 +846,17 @@ public IClientRouter getRouter(String address) {
      * If the layout has been previously invalidated and a new layout has not yet been retrieved,
      * this function does nothing.
      */
-    public synchronized void invalidateLayout() {
+    public synchronized CompletableFuture<Layout> invalidateLayout() {
```

Comment:
> pretty inconvenient and completely synchronous 
This is false. This call doesn't block until the layout is fetched. It might block on synchronized, but this method will just spawn an async task. The future generated from this method can be accessed by `CorfuRuntime::layout`

Reply:
>completely synchronous (if you want to get updated layout)

So, it is not convenient to execute two methods instead of one. 
Also, make `layout` variable public is a bad practice